packages/dbt/sodasql/dbt.py

```diff
@@ -2,9 +2,15 @@
 
 from __future__ import annotations
 
+from collections import defaultdict
+from functools import reduce
+from operator import or_
 from typing import Any
 
-from dbt.contracts.graph.compiled import ParsedModelNode, CompiledSchemaTestNode
+from dbt.contracts.graph.compiled import (
+    CompiledModelNode, CompiledSchemaTestNode
```

Comment:
note to self: double check which nodes `compiled` or `parsed` and other test besides `SchemaTest` we should expect.

Reply:
For now, I think we are ok, the dbt v1 might cause troubles, see [release notes](https://github.com/dbt-labs/dbt-core/releases/tag/v1.0.0) on tests:

> The two type of test definitions are now "singular" and "generic" (instead of "data" and "schema", respectively). The test_type: selection method accepts test_type:singular and test_type:generic. (It will also accept test_type:schema and test_type:data for backwards compatibility) (#3234, #3880). Not backwards compatible: The --data and --schema flags to dbt test are no longer supported, and tests no longer have the tags 'data' and 'schema' automatically applied.

Reply:
More general: I asked around in the [dbt slack](https://getdbt.slack.com/archives/C01F91XJ5PY/p1638266195027300) about another possible entrypoint for parsing the artifacts. Conclusion is: that using the node objects is the way to go - for now